### ISSUE TITLE
See some crashes when deleting a container

### ISSUE THREAD

**Issue by testn** [nlf/dlite#70]
Not sure whether it is something to do with btrfs or not but it looks like I cannot delete any image.

```
time="2016-01-29T20:18:14.024680000Z" level=info msg="DELETE /v1.21/containers/61762b81679a"
2016/01/29 20:18:14 http: panic serving @: runtime error: invalid memory address or nil pointer dereference
goroutine 309 [running]:
net/http.func·011()
    /usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/docker/docker/daemon/graphdriver/btrfs.func·001(0xc2093cdf00, 0x7b, 0x0, 0x0, 0x7f5f0a479bb0, 0xc20917b7a0, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/graphdriver/btrfs/btrfs.go:197 +0x63
path/filepath.Walk(0xc2093cdf00, 0x7b, 0xc20851d918, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:392 +0xb6
github.com/docker/docker/daemon/graphdriver/btrfs.subvolDelete(0xc2093cdd00, 0x72, 0xc2093cdd6a, 0x8, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/graphdriver/btrfs/btrfs.go:210 +0x236
github.com/docker/docker/daemon/graphdriver/btrfs.func·001(0xc2093cdd00, 0x72, 0x7f5f0a479bd8, 0xc208a0b630, 0x0, 0x0, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/graphdriver/btrfs/btrfs.go:203 +0x3aa
path/filepath.walk(0xc2093cdd00, 0x72, 0x7f5f0a479bd8, 0xc208a0b630, 0xc20851f148, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:347 +0x91
path/filepath.walk(0xc208886d20, 0x69, 0x7f5f0a479bd8, 0xc208a0b360, 0xc20851f148, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:372 +0x51d
path/filepath.walk(0xc208894700, 0x65, 0x7f5f0a479bd8, 0xc208a0ae60, 0xc20851f148, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:372 +0x51d
path/filepath.walk(0xc2088750a0, 0x61, 0x7f5f0a479bd8, 0xc20925f130, 0xc20851f148, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:372 +0x51d
path/filepath.Walk(0xc2088750a0, 0x61, 0xc20851f148, 0x0, 0x0)
    /usr/local/go/src/path/filepath/path.go:394 +0xf2
github.com/docker/docker/daemon/graphdriver/btrfs.subvolDelete(0xc208d26dc0, 0x20, 0xc208ce1800, 0x40, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/graphdriver/btrfs/btrfs.go:210 +0x236
github.com/docker/docker/daemon/graphdriver/btrfs.(*Driver).Remove(0xc20850ce00, 0xc208ce1800, 0x40, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/graphdriver/btrfs/btrfs.go:267 +0xf5
github.com/docker/docker/daemon/graphdriver.(*NaiveDiffDriver).Remove(0xc20850ce40, 0xc208ce1800, 0x40, 0x0, 0x0)
    <autogenerated>:29 +0x7c
github.com/docker/docker/daemon.(*Daemon).rm(0xc20806ec80, 0xc208830200, 0x0, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/delete.go:122 +0x860
github.com/docker/docker/daemon.(*Daemon).ContainerRm(0xc20806ec80, 0xc208a99ab9, 0xc, 0xc20851f625, 0x0, 0x0)
    /go/src/github.com/docker/docker/daemon/delete.go:56 +0x67a
github.com/docker/docker/api/server/router/local.(*router).deleteContainers(0xc208602080, 0x7f5f08aa7ab8, 0xc208a99da0, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/router/local/container.go:408 +0x21c
github.com/docker/docker/api/server/router/local.*router.(github.com/docker/docker/api/server/router/local.deleteContainers)·fm(0x7f5f08aa7ab8, 0xc208a99da0, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/router/local/local.go:148 +0x7b
github.com/docker/docker/api/server.func·004(0x7f5f08aa7ab8, 0xc208a99da0, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/middleware.go:87 +0x7c7
github.com/docker/docker/api/server.func·003(0x7f5f0a479a50, 0xc20802af28, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/middleware.go:66 +0x10e
github.com/docker/docker/api/server.func·002(0x7f5f0a479a50, 0xc20802af28, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/middleware.go:49 +0x47c
github.com/docker/docker/api/server.func·001(0x7f5f0a479a50, 0xc20802af28, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0, 0xc208a99b60, 0x0, 0x0)
    /go/src/github.com/docker/docker/api/server/middleware.go:27 +0x1dc
github.com/docker/docker/api/server.func·007(0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0)
    /go/src/github.com/docker/docker/api/server/server.go:156 +0x265
net/http.HandlerFunc.ServeHTTP(0xc208754840, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0)
    /usr/local/go/src/net/http/server.go:1265 +0x41
github.com/gorilla/mux.(*Router).ServeHTTP(0xc2085fa500, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0)
    /go/src/github.com/docker/docker/vendor/src/github.com/gorilla/mux/mux.go:98 +0x297
net/http.serverHandler.ServeHTTP(0xc208042900, 0x7f5f08aa7a80, 0xc208aae320, 0xc208fb60d0)
    /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc208aae280)
    /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
    /usr/local/go/src/net/http/server.go:1751 +0x35e
```

**Comment by nlf** [comment]
i've never seen that happen. if you restart the dlite service does it fix it?

**Comment by testn** [comment]
Nope it looks like a known problem in 1.9.1. See https://github.com/docker/docker/issues/14199#issuecomment-176969743

**Comment by nlf** [comment]
closing this since it's a known docker issue

**Comment by testn** [comment]
Probably it's better not to use subvolume feature?

**Comment by kujiy** [comment]
Where is the known issue which is still open? 

I have a similar problem now.

##### `docker stop` causes panic

After upgraded 1.9.1 to 1.10.1, then it started to happen.

``` bash
[root@docker /]# docker stop cent6
Failed to stop container (cent6): An error occurred trying to connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.22/containers/cent6/stop?t=10: EOF
```

```
[root@docker /]# cat /var/log/messages | grep panic
Feb 18 10:28:22 docker01 docker: panic: runtime error: index out of range
Feb 18 10:32:17 docker01 docker: panic: runtime error: index out of range
Feb 18 10:39:12 docker01 docker: panic: runtime error: index out of range
```

##### my docker infos

```
[root@docker /]#  docker version
Client:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:18:46 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.1
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   9e83765
 Built:        Thu Feb 11 19:18:46 2016
 OS/Arch:      linux/amd64
```

```
[root@docker /]#  docker info
Containers: 36
 Running: 12
 Paused: 0
 Stopped: 24
Images: 1279
Server Version: 1.10.1
Storage Driver: devicemapper
 Pool Name: docker-253:0-131271-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 51.82 GB
 Data Space Total: 107.4 GB
 Data Space Available: 55.56 GB
 Metadata Space Used: 76.03 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.071 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-327.4.5.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.44 GiB
Name: docker.**************.com
ID: FQTI:ME6T:255F:HLNF:WAH2:PVSX:BBK2:M2Q5:H5D4:IQRU:2NO6:ZZCC
Http Proxy: http://**************.com:8080
Https Proxy: http:/**************.com:8080
No Proxy: localhost,127.0.0.0/8,docker-registry.somecorporation.com
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

```
